content/en/blog/_posts/Windows-Memory-Pressure-Eviction.md

```diff
@@ -0,0 +1,85 @@
+This also means that memory limits specified for containers in Kubernetes Pods are enforced on the total amount of virtual memory committed by all processes in the container. Consistenty in how memory usage is tracked and enforced is essential for workload stability.
+
+2. **Paging (swap) is automatically enabled on Windows**, and it plays an essential role in the operating system's memory management. Disabling paging is not recommended because it can lead to system instability.
+
+## Eviction Thresholds
+
+By default, Windows nodes have a default "hard" eviction threadhold of `--eviction-hard=memory.available<500Mi`. This means that if the system's availalbe memory is less than 500 Mi the Kubelet will try and start evicting pods.
+
+For more information on configuring memory-pressure based eviction and understanding the eviction signals and threadhols, refer to the official Kubernetes documentation on [memory-pressure-eviction](https://kubernetes.io/docs/concepts/scheduling-eviction/node-pressure-eviction/#eviction-signals-and-thresholds).
+
+
+## Monitoring Commit Memory for Windows nodes
+
+Understanding the distinction between **Working Set** and **Commit** memory and being able to monitor both is critical for maintaining the stability of individual Pod workloads and the entire Windows node.
+
+Currently, `kubectl top` continues to report Working Set memory. As a result, a Pod may be evicted for exceeding its memory requests or encounter memory allocation failures due to exceeding its Commit limit, even when the WorkingSet appears to be well within the expected bounds. This discrepancy can lead to confusion.
+
+Starting with v1.31, Windows nodes now report the global `CommitTotal` and `CommitLimit` for each node through the `/stats/summary` API endpoint. This information is available under the `windows-global-commit-memory` syscontainer.
+
+```json=
+kubectl get --raw http://{cluster-endpoint}/api/v1/nodes/{node-name}/proxy/stats/summary
+
+{
+ "node": {
+  "nodeName": "{node-name},
+  "systemContainers": [
+   ...
+   {
+    "name": "windows-global-commit-memory",
+    "startTime": "2024-10-02T19:14:33Z",
+    "memory": {
+     "time": "2024-10-02T19:14:33Z",
+     "availableBytes": 4563402752,
+     "usageBytes": 1212754432
+    }
+   }
+  ...
+  ],
+}
+
+```
+
+
+Additionally, exposing the Commit usage for individial Pods and Containers through the `/stats/summary` API is being worked on as part of the [cAdvisor-les, CRI-full Container and Pod Stats](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/2371-cri-pod-container-stats/README.md) enhancement.
+
+[windows-exporter](https://github.com/prometheus-community/windows_exporter) can also be configured to monitor and report on Commit memory usage.
```

Comment:
maybe mention it works for containers and node level today if that information is required now?

Reply:
updated - PTAL